File: algorithms/js/countPrimes/countPrimes.js
// Source : https://leetcode.com/problems/count-primes/
// Author : Harvey Cao
// Date   : 2021-05-10

/********************************************************************************** 
* 204. Count Primes
* Count the number of prime numbers less than a non-negative number, n.
*               
**********************************************************************************/

/**
 * @param {number} n
 * @return {number}
 */
 var countPrimes = function(n) {
    const primes = new Array(n).fill(true);
    let res = 0;
    for (let i = 2; i < n; i++) {
        if (primes[i]) {
            res++;
            for (let j = i * i; j < n; j += i) primes[j] = false;
        }
    }
    return res;
};

File: algorithms/python/countPrimes/countPrimes.py
'''
* 204. Count Primes
* Count the number of prime numbers less than a non-negative number, n.
'''

class Solution:
    def countPrimes(self, n: int) -> int:
        primes = [True] * n
        res = 0
        for i in range(2, n):
            if primes[i]:
                res += 1
                for j in range(i * i, n, i):
                    primes[j] = False
        return res